### PR TITLE
[Item Charges] Add Amulet of Chemistry and Bounty

### LINKED ISSUE #9246: Amulet of Chemistry/similar item replace overlay
**Is your feature request related to a problem? Please describe.**
When I'm making potions I'm always having to check my equipment spot to see if the amulet has run out. This is because when I'm making prayer potions for example if the amulet breaks on the 14th slot it's hard to tell whether the amulet needs replacing without checking equipment, because the message flies by the chatbox.
 
**Describe the solution you'd like**
An overlay telling me I need to replace the amulet. I know other amulets/rings have similar effects, so it would be nice if they could get similar treatment too.

### LINKED ISSUE #10781: Add amulet of bounty to item charges plugin
Add the amulet of bounty to the item charges plugin.

Below are the text strings you should need to reference. Apologies for the mobile screenshots, away from my PC for a few days.

Charge being used:
3838636F-7B44-4AAC-82A8-C57BAA63D65D

Using the ‘check’ option on the amulet
3E1A565F-C975-492E-9082-58E40D0CA159

One charge remaining:
DC189768-8DC6-495E-A518-7FF43B8B2FDB

Using the last charge:
2BA25387-0016-4AC6-8D7E-DEFCA2636572

Manually breaking the amulet using the option to:
(Doesn’t show up in the chat box, just this pop up)
2ECD18D5-155C-45AC-8C7B-68F737E6FDAF

### PR DESCRIPTION
**Pull request by dabolink**
Adds Tracker for Amulet of Chemistry Charges and Amulet of Bounty Charges
image

Closes #9246
Closes #10781

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 1]
```diff
@@ -80,11 +80,19 @@
 		"You can smelt ([0-9+]+|one) more pieces? of iron ore before a ring melts\\.");
 	private static final String RING_OF_FORGING_USED_TEXT = "You retrieve a bar of iron.";
 	private static final String RING_OF_FORGING_BREAK_TEXT = "<col=7f007f>Your Ring of Forging has melted.</col>";
+	private static final Pattern AMULET_OF_CHEMISTRY_CHECK_PATTERN = Pattern.compile(
+		"Your amulet of chemistry has (\\d+) charges? left\\."
+	);
+	private static final Pattern AMULET_OF_CHEMISTRY_USED_PATTERN = Pattern.compile(
+		"Your amulet of chemistry helps you create a 4-dose potion. (?:<col=ff0000>)?It has ([0-9]+|one) charges? left.(?:</col)?"
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java]: Is the trailing `(?:</col)` group necessary?

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 2]
```diff
@@ -80,11 +80,19 @@
 		"You can smelt ([0-9+]+|one) more pieces? of iron ore before a ring melts\\.");
 	private static final String RING_OF_FORGING_USED_TEXT = "You retrieve a bar of iron.";
 	private static final String RING_OF_FORGING_BREAK_TEXT = "<col=7f007f>Your Ring of Forging has melted.</col>";
+	private static final Pattern AMULET_OF_CHEMISTRY_CHECK_PATTERN = Pattern.compile(
+		"Your amulet of chemistry has (\\d+) charges? left\\."
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java]: If 5 is the max number of charges for an amulet of chemistry, why does this accept `\\d+`?

Comment (dabolink) [reply runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java]: you're right. probably a copy paste error

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java (hunk 1 above)
Comment (dabolink) [reply runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java]: was trying to match the entire thing. but probably not needed and does clean it up a bit

File: runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java [hunk 3]
```diff
@@ -80,11 +80,19 @@
 		"You can smelt ([0-9+]+|one) more pieces? of iron ore before a ring melts\\.");
 	private static final String RING_OF_FORGING_USED_TEXT = "You retrieve a bar of iron.";
 	private static final String RING_OF_FORGING_BREAK_TEXT = "<col=7f007f>Your Ring of Forging has melted.</col>";
+	private static final Pattern AMULET_OF_CHEMISTRY_CHECK_PATTERN = Pattern.compile(
+		"Your amulet of chemistry has (\\d) charges? left\\."
+	);
+	private static final Pattern AMULET_OF_CHEMISTRY_USED_PATTERN = Pattern.compile(
+		"Your amulet of chemistry helps you create a 4-dose potion. (?:<col=ff0000>)?It has ([0-9]+|one) charges? left."
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java]: The last character should be a literal period, not any character.
```suggestion
		"Your amulet of chemistry helps you create a 4-dose potion. (?:<col=ff0000>)?It has ([0-9]+|one) charges? left\\."
```

**Comment by dabolink** [comment]
I thought I rebased. but I guess I accidentally did a merge. 🤦‍♂

**Comment by Nightfirecat** [comment]
@dabolink I've squashed and force-pushed to your branch. These changes look good to go now.

**Comment by dabolink** [comment]
Added Tracking of Amulet of Bounty as well

**Comment by Adam-** [comment]
@Nightfirecat can you fixup the second bounty commit now too?

**Pull request merged**